app/models/P5/departure/MessageData.scala

```diff
@@ -17,46 +17,56 @@
 package models.P5.departure
 
 import play.api.libs.json.Json
-import play.api.libs.json.OWrites
 import play.api.libs.json.Reads
-import play.api.libs.json.Writes
 
-case class DepartureMessageData(
-  TransitOperation: TransitOperation,
-  HolderOfTheTransitProcedure: HolderOfTransitProcedure,
-  Representative: Representative,
-  Consignment: Consignment,
-  Guarantee: Option[List[Guarantee]],
+sealed trait MessageData
+
+case class IE015MessageData(TransitOperation: IE015TransitOperation) extends MessageData
+
+object IE015MessageData {
+  implicit val reads: Reads[IE015MessageData] = Json.reads[IE015MessageData]
+}
+
+case class IE029MessageData(
+  TransitOperation: IE029TransitOperation,
   Authorisation: Option[List[Authorisation]],
+  CustomsOfficeOfDeparture: CustomsOfficeOfDeparture,
+  CustomsOfficeOfDestinationDeclared: CustomsOfficeOfDestinationDeclared,
   CustomsOfficeOfTransitDeclared: Option[List[CustomsOfficeOfTransitDeclared]],
   CustomsOfficeOfExitForTransitDeclared: Option[List[CustomsOfficeOfExitForTransitDeclared]],
-  CustomsOfficeOfDeparture: CustomsOfficeOfDeparture,
-  CustomsOfficeOfDestinationDeclared: CustomsOfficeOfDestinationDeclared
-) {
+  HolderOfTheTransitProcedure: HolderOfTransitProcedure,
+  Representative: Representative,
+  ControlResult: Option[ControlResult],
+  Guarantee: Option[List[Guarantee]],
+  Consignment: Consignment
+) extends MessageData {
 
-  val guarantee: Option[String] = Guarantee.map(
+  val guaranteeDisplay: Option[String] = Guarantee.map(
     _.map(_.toString).mkString("; ")
   )
 
-  val authorisation: Option[String] = Authorisation.map(
-    _.map(_.toString).mkString("; ")
+  val authorisationDisplay: Option[String] = Authorisation.map(
+    authorisations =>
+      authorisations.length match {
+        case 0 => ""
```

Comment:
could this have been done for `departureTransportMeansIdentity` too? The one in app/models/P5/departure/Consignment.scala